Delay docsig function call with timer

jdtsmith opened the pull request:
This implements a timer for calling the company-docsig function and displaying its results.  The impetus is to work more smoothly with costly docsig function which interact with servers or processes.    This uses only a single short delay, which I find works well, allowing you to scroll through many candidates without querying for docsig, then pausing briefly on any candidate to query and echo its short documentation.  Fixes #131.

minad commented:
This doesn't work well for me. While I like a secondary short delay of 0.2s I wouldn't want an initial delay of only 0.2s. Therefore two delays.

The question is what to do if there is a visible docsig and the candidate changed. Immediately hide the message? Otherwise you will see an outdated docsig in the echo area.

jdtsmith commented:
I immediately hide message on new candidate, so you are never out of date.  For very short delays (~0.1s) this can lead to flashing as you arrow through candidates. 

I have implemented an initial/subsequent delay, either of which can be `t` (for instant).  Sounds like you may like `(1.0 t)` or similar.  I added a state variable `corfu--initial` to keep track of whether this is the first update.

minad commented:
Thanks, I simplified it a little bit in 7f5dacb340f0ec3f069744a76b3021b055ac98ed.
